python/graphscope/client/session.py

```diff
@@ -781,6 +781,21 @@ def __del__(self):
         except Exception:  # pylint: disable=broad-except
             pass
 
+    def __enter__(self):
+        """Register self as default session"""
+        self.as_default()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        """Deregister self from the default session,
+        close the session and release the resources.
+        """
+        try:
+            self._deregister_default()
+            self.close()
+        except Exception:
+            pass
```

Comment:
No need to this catch, as we already use `except Exception` to catch all in `self.close()` itself.

Reply:
No, `self.close()` doesn't do `except Exception`, it's in `__del__`

Reply:
I see, then it looks good to me.